### DISCUSSION TITLE
Potentially run cleanup tasks once "use this template" button is clicked

### DISCUSSION THREAD

**Discussion by qlonik** [onedr0p/cluster-template#1532] [Ideas]
Stumbled upon the template project, which has some github workflow files that clean up repo after it is initialized. 

Are there some inspiration ideas for this repo to do something similar?

- https://github.com/JetBrains/intellij-platform-plugin-template/blob/main/.github/workflows/template-cleanup.yml
- https://github.com/JetBrains/intellij-platform-plugin-template/blob/main/.github/workflows/template-verify.yml

**Comment by onedr0p** [comment]
Not a bad idea, but I wonder if this has some unattended outcomes. People might not be expecting this.

**Reply by qlonik** [reply]
That's true, might be unexpected at first. Although renovatebot could also do some automation tasks, that maybe unexpected for inexperienced people. Readme could point out this behavior. But at least some tests that only make sense within the template repo can be removed this way.

Might be unexpected in the other direction too - after getting used to something cleaning up when clicking "use this template" button, now a simple repo clone won't remove those setup scripts :shrug: 

Mainly wanted to bring attention to this automation possibility, while I don't have a clear understanding if it applies here :)
